### ISSUE TITLE
Non administrators can't register for events

### ISSUE THREAD

**Issue by larspetrus** [cubing/ccm#209]
I20150303-12:29:21.214(-8)? Exception while invoking method '/registrations/insert' ReferenceError: fields is not defined
I20150303-12:29:21.214(-8)?     at Registrations.allow.insert (app/both/access_control.js:290:23)

**Comment by jfly** [comment]
Wow, that's odd. It looks like the `insert` function signature on https://github.com/cubing/ccm/blob/master/both/access_control.js#L276 needs a 3rd argument named `fields`, just like the `update` below it https://github.com/cubing/ccm/blob/master/both/access_control.js#L295.

**Comment by larspetrus** [comment]
The insert function only get sent those 2 parameters (http://docs.meteor.com/#/full/allow).

So if you can insert, you can set all fields. You have to be able to set competitionId and userId, for example.

The other option, if we want to ensure valid foreign keys, is to insert records through Meteor methods rather than Mongo.

**Comment by jfly** [comment]
Ah, thanks. I did not realize this is an insert method. In that case, I believe the minimal fix here is to change [line 290](https://github.com/cubing/ccm/blob/e20f64b7e020f30c41abc30d7748716c485c5945/both/access_control.js#L290) from

```javascript
if(_.difference(fields, allowedOwnRegistrationFields).length > 0) {
```

to

```javascript
if(_.difference(Object.keys(registration), allowedOwnRegistrationFields).length > 0) {
```

I do find methods to generally be easier to write and easier to read than access rules like these, but in this case, the fix I've described seems simple enough.
